projects/plugins/jetpack/_inc/client/at-a-glance/stats/index.jsx

```diff
@@ -77,14 +77,19 @@ export class DashStats extends Component {
 				chartLabel = '',
 				tooltipLabel = '';
 
+			// Dates from API are GMT so we want to make sure that is properly observed.
+			if ( 'week' === unit ) {
+				date = date.replace( /W/g, '-' );
+			}
+			date = new Date( `${ date }T00:00:00Z` ).toISOString();
```

Comment:
Nice fix! This is something we have been dealing with for a while. Very confusing Date constructor 😢